### DISCUSSION TITLE
I can't "pip install black"

### DISCUSSION THREAD

**Discussion by Forealthy** [smartcontractkit/full-blockchain-solidity-course-py#1601] [Q&A] [answered]
Hello, when I'm installing black with "pip install black", it throws some errors, I put images below. Is there any friends having faced this situation?

images:
![pip_black1](https://user-images.githubusercontent.com/109839460/180724147-0710755c-d7fe-4ed3-8c79-2593f1597fe4.png)
... ... tons of errors ... ...
![pip_black2](https://user-images.githubusercontent.com/109839460/180724165-47c70c1d-b7dc-47d0-b054-675dd467bbb1.png)
... ... tons pf errors ... ...
![pip_black3](https://user-images.githubusercontent.com/109839460/180724191-037bc029-baa3-4a56-8592-67f726bb858d.png)

Anyone can help? Thanks a lot!

**Comment by cromewar** [comment, marked as answer]
Hello @Forealthy 
well black is a common tool used not just for web3 projects but for python development in general, based on the error it seems to have troubles downloading the installation script: in my experience this is most likely due to the Python version you are using, I strongly suggest to try Python version 3.8 instead.

**Comment by fantauzzi** [comment]
I know it is not what you asked, but you can also try PEP8. If you use vscode, if you have no formatter configured for Python, the first time you try to format Python source code, vscode will ask you if you want to install it.

**Reply by cromewar** [reply]
this is also a good option.
